addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/script/ScriptBasedEncodeDecodeProcessor.java

```diff
@@ -75,7 +75,10 @@ public EncodeDecodeResult process(String value) throws Exception {
             EncodeDecodeScript script = evaluateScript(scriptWrapper);
             if (script != null) {
                 LOGGER.debug("Calling encode/decode script {}", scriptWrapper.getName());
-                return script.process(value);
+                Object result = script.process(value);
+                return result instanceof EncodeDecodeResult
+                        ? (EncodeDecodeResult) result
+                        : new EncodeDecodeResult((String) result);
```

Comment:
This needs tests (it will fail if the return type is neither `EncodeDecodeResult` nor `String`).

Reply:
I'll see what I can do. Might need an example.

Note that was already the case, this is just ever so slightly more forgiving.

Reply:
Do we have an existing example of testing unittesting scripts?

Reply:
There seem to be some related things in the scripts add-on related to the automation add-on, but it isn't actually executing anything it's just using mocks.

Reply:
In the meantime, shall I wrap this in a try/catch and throw an IOException stating the return was neither a EncodeDecodeResult nor String?

Reply:
While it failed before it's not the same case, before it would fail in the script engine, now fails in our code.

The test in `VariantCustomUnitTest`, you don't need to test script engines, any class that implements/mocks the required interface allows to test this code.

Reply:
It's preferable to use `String.valueOf(result)` instead of casting to `String` or throwing an exception.

Reply:
I have changed this to `String.valueOf(result)`, but I still need to work on tests.

Reply:
Okay, I've added two tests for the existing implementation. They're kinda weak but it gives me something to build on while I continue the interface piece.

Reply:
It should also test with other type (which would trigger the exception).

Reply:
Hmmmm, how were you picturing that since it's not actually processing things, an object is returned so an object is returned 🤷‍♂️ 

```js
    @Test
    void shouldThrowOnUnexpectedReturnType() throws Exception {
        // Given
        ScriptWrapper scriptWrapper = mock(ScriptWrapper.class);
        given(scriptWrapper.isEnabled()).willReturn(true);
        ExtensionScript extScript = mock(ExtensionScript.class);
        EncodeDecodeScript script = mock(EncodeDecodeScript.class);
        given(extScript.getInterface(scriptWrapper, EncodeDecodeScript.class)).willReturn(script);
        String expected = "admin";
        given(script.process("admin")).willReturn(new Random());
        // When
        Object result = script.process("admin");
        // Then
        System.out.println(result.getClass());
//        assertTrue(result instanceof String);
//        assertThat(result, is(equalTo(expected)));
    }
```

What would my assertion be, since it's all mocked there's no exception.

Actually I did the above as an actual script and now that we're using `String.valueof` it depends on the `toString()` of the returned object, so I actually got `java.util.Random@40b8b10d`.

```js
var Random = Java.type("java.util.Random");

function process(value){
	return new Random();
}
```

Reply:
Would trigger the exception with the previous code, now it's properly handled and converted to string. Use an int or other type which implements `toString()`.

Reply:
Don’t we want it to actually cause an exception though, so that the user is aware they’ve returned the wrong thing?

Reply:
If we are allowing to return anything seems ok to me to accept anything, but as long as the behaviour is documented I don't really mind what it is.

Reply:
Okay, let’s stick with this. We can always change down the line with user feedback.